tests/gocase/unit/type/bloom/bloom_test.go

```diff
@@ -145,6 +146,82 @@ func TestBloom(t *testing.T) {
+
+		require.Equal(t, []interface{}{int64(1), int64(1)}, rdb.Do(ctx, "bf.madd", key, "xxx", "zzz").Val())
+		require.Equal(t, int64(2), rdb.Do(ctx, "bf.card", key).Val())
+		require.Equal(t, []interface{}{int64(1), int64(0), int64(1)}, rdb.Do(ctx, "bf.mexists", key, "xxx", "yyy", "zzz").Val())
+
+		// add the existed value
+		require.Equal(t, []interface{}{int64(0)}, rdb.Do(ctx, "bf.madd", key, "zzz").Val())
+		require.Equal(t, []interface{}{int64(1), int64(0), int64(1)}, rdb.Do(ctx, "bf.mexists", key, "xxx", "yyy", "zzz").Val())
+
+		// add the same value
+		require.Equal(t, []interface{}{int64(1), int64(0)}, rdb.Do(ctx, "bf.madd", key, "yyy", "yyy").Val())
+		require.Equal(t, []interface{}{int64(1), int64(1), int64(1)}, rdb.Do(ctx, "bf.mexists", key, "xxx", "yyy", "zzz").Val())
+	})
+
+	t.Run("MAdd nonscaling Test", func(t *testing.T) {
+		require.NoError(t, rdb.Del(ctx, key).Err())
+		require.NoError(t, rdb.Do(ctx, "bf.reserve", key, "0.0001", "25", "nonscaling").Err())
+
+		// insert items, suppose false positives is 0
+		require.Equal(t, []interface{}{int64(1), int64(1), int64(1), int64(1)}, rdb.Do(ctx, "bf.madd", key, "x", "y", "z", "k").Val())
+		for i := 0; i < 10; i++ {
+			buf := util.RandString(7, 8, util.Alpha)
+			Add := rdb.Do(ctx, "bf.madd", key, buf, buf+"xx")
+			require.NoError(t, Add.Err())
+		}
+		require.Equal(t, int64(24), rdb.Do(ctx, "bf.card", key).Val())
+
+		Add := rdb.Do(ctx, "bf.madd", key, "a", "x", "xxx", "y", "z").Val()
+		ret := make([]interface{}, 0, 5)
+		for _, value := range Add.([]interface{}) {
+			switch v := value.(type) {
+			case int64:
+				ret = append(ret, v)
+			case error:
+				ret = append(ret, v.Error())
+			default:
+			}
+		}
+		assert.Equal(t, []interface{}{int64(1), int64(0), "ERR nonscaling filter is full", int64(0), int64(0)}, ret)
+		require.Equal(t, int64(25), rdb.Do(ctx, "bf.card", key).Val())
+	})
+
+	t.Run("MAdd scaling Test", func(t *testing.T) {
+		require.NoError(t, rdb.Del(ctx, key).Err())
+		require.NoError(t, rdb.Do(ctx, "bf.reserve", key, "0.0001", "30", "expansion", "2").Err())
+
+		// insert items, suppose false positives is 0
+		for i := 0; i < 10; i++ {
+			buf := util.RandString(7, 8, util.Alpha)
+			Add := rdb.Do(ctx, "bf.madd", key, buf, buf+"xx", buf+"yy")
+			require.NoError(t, Add.Err())
+		}
+		require.Equal(t, []interface{}{"Capacity", int64(30), "Size", int64(128), "Number of filters", int64(1), "Number of items inserted", int64(30), "Expansion rate", int64(2)}, rdb.Do(ctx, "bf.info", key).Val())
+
+		// bloom filter is full and scaling
+		require.NoError(t, rdb.Do(ctx, "bf.add", key, "xxx").Err())
+		require.Equal(t, []interface{}{"Capacity", int64(90), "Size", int64(384), "Number of filters", int64(2), "Number of items inserted", int64(31), "Expansion rate", int64(2)}, rdb.Do(ctx, "bf.info", key).Val())
+
+		// insert items, suppose false positives is 0
```

Comment:
Would this produce some flacky test?

Reply:
Sure, It's maybe flacky with a very small probability. I will fix it.